server/src/ast.ts

```diff
@@ -511,4 +525,164 @@ export type ConcreteType =
     | StructType
     | { tag: "ArrayType" }
     | { tag: "PointerType" }
-    | { tag: "TaggedPointerType" };
\ No newline at end of file
+    | { tag: "TaggedPointerType" };
+
+export type Ordering = 
+  | "Less"
+  | "Equal"
+  | "Greater"; 
```

Comment:
I think this should be a const enum rather than strings. Or, you could make it integer enums (so you could compare them with <, like order <= Ordering.Equal instead of comparing individually)

Reply:
Good catch, for some reason I totally forgot enums existed